framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -120,24 +120,44 @@
     private static final int PARAMETER_REPLACEMENT_LENGTH = PARAMETER_REPLACEMENT.length();
 
     /**
-     * Parse a string and return its representation as a {@link JavaExpression}, or throw a {@link
+     * Parse a string and viewpoint-adapt it to the given {@code context}. Return its representation
+     * as a {@link JavaExpression}, or throw a {@link JavaExpressionParseException}.
+     *
+     * @param expression a Java expression to parse
+     * @param context information about any receiver and arguments
+     * @return the JavaExpression for the given string
+     * @throws JavaExpressionParseException if the string cannot be parsed
+     */
+    public static JavaExpression parse(String expression, JavaExpressionContext context)
+            throws JavaExpressionParseException {
+        return parse(expression, context, null);
+    }
+    /**
+     * Parse a string and viewpoint-adapt it to the given {@code context} and {@code localPath}.
```

Comment:
I don't think that viewpoint-adaptation is with respect to a given path.  Should this be "parse with respect to localPath and viewpoint-adapt with respect to context"?